src/databricks/labs/ucx/azure/resources.py

```diff
@@ -140,6 +140,15 @@ class AzureRoleAssignment:
     scope: AzureResource
     principal: Principal
     role_name: str
+    role_type: str
+    role_permissions: list
+
+
+@dataclass
+class AzureRoleDetails:
+    role_name: str | None
+    role_type: str
+    role_permissions: list
```

Comment:
```suggestion
    role_permissions: list[str]
```

Reply:
@JCZuurmond - Thanks, I have added this.